plugins/modules/k8s_cluster_info.py

```diff
@@ -0,0 +1,185 @@
+      returned: success
+      type: bool
+    prefix:
+      description: Resource Path Prefix
+      returned: success
+      type: str
+    short_names:
+      description: Resource short names
+      returned: success
+      type: str
+    singular_name:
+      description: Resource singluar name
+      returned: success
+      type: str
+    urls:
+      description: Resource urls
+      returned: success
+      type: dict
+    available_api_version:
+      description: All available versions of the given API
+      returned: success
+      type: list
+    preferred_api_version:
+      description: Preferred version of the given API
+      returned: success
+      type: str
+'''
+
+
+import copy
+import traceback
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.kubernetes.plugins.module_utils.common import K8sAnsibleMixin, AUTH_ARG_SPEC
+try:
+    from kubernetes import client
+except ImportError:
+    # ImportError are managed by the common module already.
+    pass
+
+try:
+    try:
+        # >=0.10
+        from openshift.dynamic.resource import ResourceList
+    except ImportError:
+        # <0.10
+        from openshift.dynamic.client import ResourceList
+    HAS_K8S_INSTANCE_HELPER = True
+    k8s_import_exception = None
+except ImportError:
+    HAS_K8S_INSTANCE_HELPER = False
+    k8s_import_exception = traceback.format_exc()
+
+
+class KubernetesInfoModule(K8sAnsibleMixin):
+
+    def __init__(self):
+        module = AnsibleModule(
+            argument_spec=self.argspec,
+            supports_check_mode=True,
+        )
+        self.module = module
+        self.params = self.module.params
+
+        if not HAS_K8S_INSTANCE_HELPER:
+            self.module.fail_json(msg=missing_required_lib("openshift >= 0.6.2", reason="for merge_type"),
+                                  exception=k8s_import_exception)
+
+        super(KubernetesInfoModule, self).__init__()
+
+    def execute_module(self):
+        self.client = self.get_api_client()
+        results = {}
+        for resource in list(self.client.resources):
+            resource = resource[0]
+            if isinstance(resource, ResourceList):
+                continue
+            results[resource.group] = {
+                'api_version': resource.api_version,
+                'categories': resource.categories if resource.categories else [],
+                'group_version': resource.group_version,
+                'kind': resource.kind,
+                'name': resource.name,
+                'namespaced': resource.namespaced,
+                'preferred': resource.preferred,
+                'prefix': resource.prefix,
+                'short_names': resource.short_names if resource.short_names else [],
+                'singular_name': resource.singular_name,
+                'urls': resource.urls,
+            }
+
+        for api in client.ApisApi().get_api_versions().groups:
```

Comment:
we can probably skip this, we already store whether a resource is `preferred` or not which should be enough to filter on if the information is needed